stub/src/main/java/io/grpc/stub/StreamObserver.java

```diff
@@ -43,9 +47,17 @@
    * server streaming calls, but may receive many onNext callbacks.  Servers may invoke onNext at
    * most once for client streaming calls, but may receive many onNext callbacks.
    *
-   * <p>If an exception is thrown by an implementation the caller is expected to terminate the
-   * stream by calling {@link #onError(Throwable)} with the caught exception prior to
-   * propagating it.
+   * <p>Although implementations are expected to not throw exceptions, if an exception occurs the
+   * caller is encouraged to call {@link #onError} as part of normal application exception handling.
+   * That is to say, {@code onError} should be called if any related part of the application throws
+   * an exception, generally independent of the source, to avoid leaks. gRPC will call {@code
+   * onError()} when an application's implementation throws.
```

Comment:
This is helpful but perhaps it should go in the javadoc for the whole interface since it isn't specific to onNext? You could say:

Owners of a StreamObserver are responsible for eventually calling either onError() or onCompleted() to ensure that  resources associated with the stream (memory, file descriptors, etc) are released.